SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -408,6 +421,125 @@ void FSpatialGDKEditorToolbarModule::OnPropertyChanged(UObject* ObjectBeingModif
 	}
 }
 
+bool FSpatialGDKEditorToolbarModule::GenerateDefaultLaunchConfig(const FString& LaunchConfigPath) const
+{
+	FString Text;
+	TSharedRef< TJsonWriter<> > Writer = TJsonWriterFactory<>::Create(&Text);
+
+	// Populate json file for launch config
+	Writer->WriteObjectStart(); // Start of json
+	Writer->WriteValue(TEXT("template"), TEXT("small")); // Template section
+	Writer->WriteObjectStart(TEXT("world")); // World section begin
+	Writer->WriteObjectStart(TEXT("dimensions"));
+	Writer->WriteValue(TEXT("x_meters"), 2000);
+	Writer->WriteValue(TEXT("z_meters"), 2000);
+	Writer->WriteObjectEnd();
+	Writer->WriteValue(TEXT("chunk_edge_length_meters"), 50);
+	Writer->WriteValue(TEXT("streaming_query_interval"), 4);
+	Writer->WriteArrayStart(TEXT("legacy_flags"));
+	Writer->WriteObjectStart();
+	Writer->WriteValue(TEXT("name"), TEXT("streaming_query_diff"));
+	Writer->WriteValue(TEXT("value"), TEXT("true"));
+	Writer->WriteObjectEnd();
+	Writer->WriteObjectStart();
+	Writer->WriteValue(TEXT("name"), TEXT("bridge_qos_max_timeout"));
+	Writer->WriteValue(TEXT("value"), TEXT("0"));
+	Writer->WriteObjectEnd();
+	Writer->WriteObjectStart();
+	Writer->WriteValue(TEXT("name"), TEXT("bridge_soft_handover_enabled"));
+	Writer->WriteValue(TEXT("value"), TEXT("false"));
+	Writer->WriteObjectEnd();
+	Writer->WriteObjectStart();
+	Writer->WriteValue(TEXT("name"), TEXT("qos_max_unacked_pings_rate"));
+	Writer->WriteValue(TEXT("value"), TEXT("10"));
+	Writer->WriteObjectEnd();
+	Writer->WriteArrayEnd();
+	Writer->WriteObjectStart(TEXT("snapshots"));
+	Writer->WriteValue(TEXT("snapshot_write_period_seconds"), 0);
+	Writer->WriteObjectEnd();
+	Writer->WriteObjectEnd(); // World section end
+	Writer->WriteObjectStart(TEXT("load_balancing")); // Load balancing section begin
+	Writer->WriteArrayStart("layer_configurations");
+	Writer->WriteObjectStart();
+	Writer->WriteValue(TEXT("layer"), TEXT("UnrealWorker"));
+	Writer->WriteObjectStart("rectangle_grid");
+
+	int Cols = 1;
+	int Rows = 1;
+
+	if (const ULevelEditorPlaySettings* PlayInSettings = GetDefault<ULevelEditorPlaySettings>())
+	{
+		int NumServers = 1;
+		PlayInSettings->GetPlayNumberOfServers(NumServers);
+		
+		if (NumServers <= 2)
+		{
+			Cols = NumServers;
+			Rows = 1;
+		}
+		else
+		{
+			//Find greatest divisor.
+			int GreatestDivisor = 1;
+			for (int Divisor = 2; Divisor <= FMath::Sqrt(NumServers); Divisor++)
```

Comment:
uber nit: if you started at sqrt(NumServer) and moved to 2, you could early out when the first modular operation succeeds.